printer_tray/ir_report.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Yannick Vaucher
+#    Copyright 2013 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields
+
+
+class IrActionsReportXml(models.Model):
+
+    _inherit = 'ir.actions.report.xml'
+
+    printer_tray_id = fields.Many2one(
```

Comment:
No, the tray field can be empty and means that cups will print on the default tray of the printer. It is preferable to leave the field empty rather than copy this value from the cups server, because if the default tray is changed on the cups server, the tray stored in the records would no longer be correct.